lua/dependency_assist/api.lua

```diff
@@ -12,12 +12,17 @@ end
 --- @param path string
 --- @param cb function
 --- @return number 'the job id'
-function Api:get(path, cb)
+function Api:get(path, cb, provider_header)
+  if provider_header ~= nil then
+    header = provider_header
+  else
+    header = '"Content-Type: application/json"'
```

Comment:
Rather than this if you changed the arguments to a table as I mentioned above then the application/json part could be the default value e.g.
```lua
local header = opts.header or 'Content-type: application/json'
```